endpoints/openrtb2/video_auction.go

```diff
@@ -255,9 +255,9 @@ func handleError(labels pbsmetrics.Labels, w http.ResponseWriter, errL []error,
 		}
 		errors = fmt.Sprintf("%s %s", errors, er.Error())
 	}
-	w.WriteHeader(status)
+	(*w).WriteHeader(status)
 	ao.Status = status
-	fmt.Fprintf(w, "Critical error while running the video endpoint: %v", errors)
+	fmt.Fprintf(*w, "Critical error while running the video endpoint: %v", errors)
 	glog.Errorf("/openrtb2/video Critical error: %v", errors)
 	ao.Errors = append(ao.Errors, errL...)
```

Comment:
The pointers you added are a great catch but I have a question. I tried the following test:
```
718 func TestHandleError(t *testing.T) {
719     testHandler := func(w http.ResponseWriter, r *http.Request) {
720         fmt.Println("[TestHandleError] function start")
721         labels := pbsmetrics.Labels{
722             Source:        pbsmetrics.DemandUnknown,
723             RType:         pbsmetrics.ReqTypeVideo,
724             PubID:         pbsmetrics.PublisherUnknown,
725             Browser:       pbsmetrics.BrowserSafari,
726             CookieFlag:    pbsmetrics.CookieFlagUnknown,
727             RequestStatus: pbsmetrics.RequestStatusOK,
728         }
729         err := &errortypes.BadInput{Message: "Unable to find required stored request id"}
730         ao := analytics.AuctionObject{
731             Status: http.StatusOK,
732             Errors: make([]error, 0),
733         }
734
735         handleError(&labels, &w, []error{err}, &ao)
736
737         fmt.Printf("[TestHandleError] print labels.RequestStatus = %v \n", labels.RequestStatus)
738         fmt.Printf("[TestHandleError] print ao.Status = %v \n", ao.Status)
739         fmt.Printf("[TestHandleError] print ao.Errors = %v \n", ao.Errors)
740         fmt.Printf("[TestHandleError] print w.Header() = %v \n", w.Header())
741         fmt.Println("[TestHandleError] function end")
742         return
743     }
744     http.HandleFunc("/testEndpoint", testHandler)
745     log.Fatal(http.ListenAndServe(":8080", nil))
746 }
video_auction_test.go 
```

Reply:
and I get the following:
```
╰─➤  dlv test
# github.com/prebid/prebid-server/endpoints/openrtb2 [github.com/prebid/prebid-server/endpoints/openrtb2.test]
```

Reply:
Googled this thing and found a post in stackoverflow (https://stackoverflow.com/questions/22157514/passing-http-responsewriter-by-value-or-reference?rq=1) mentioning interfaces are pointers and no pointer is needed in functions similar to `handleError`. Did you get different results?

Reply:
Ah I actually did not catch that interfaces do not need explicit pointers. I'll update this in a moment to remove that as it is not necessary from the info you posted.

Reply:
Removed the pointer to the ResponseWriter. Thank you for catching that!